### ISSUE TITLE
User installed apps for file viewing/editing

### ISSUE THREAD

**Issue by maks** [maks/MGit#316]
MGit needs to use standard android intents for viewing editing files rather then shipping its own builtin viewer/editor. For case where none are installed, a very basic text viewer will still be provided by MGit.
This will allow richer viewer functionality for any file type, as was attempted in #314 without having to build all this support into MGit itself.

**Comment by kaeptmblaubaer1000** [comment]
But then the basic editor (which is much better than the default that I have, e.g. highlighting) still needs to be available.
And another point: where do you want to display the commits for the file then?

**Comment by maks** [comment]
@kaeptmblaubaer1000 Yes good point. Sorry was in a hurry yesterday. Yes the idea is that the current plain text file view, with the commits tab next to it will stay, but would add a "view" alongside the existing "edit in other apps" which would just become "edit" and editing with the internal editor would no longer be available. I think the existing editor is actually quite useful so I plan to split it off into its own seperate project and for that would welcome a markdown preview contribution like #314
